include/power_grid_model/math_solver/iterative_current_pf_solver.hpp

```diff
@@ -0,0 +1,182 @@
+// SPDX-FileCopyrightText: 2022 Contributors to the Power Grid Model project <dynamic.grid.calculation@alliander.com>
+//
+// SPDX-License-Identifier: MPL-2.0
+
+// Check if the name means anything
+#pragma once
+#ifndef POWER_GRID_MODEL_MATH_SOLVER_ITERATIVE_CURRENT_PF_SOLVER_HPP
+#define POWER_GRID_MODEL_MATH_SOLVER_ITERATIVE_CURRENT_PF_SOLVER_HPP
+// To avoid unused parameter and variable error
+#define UNUSED(x) (void)(x)
+
+/*
+Iterative Power Flow
+I = YU
+Split this with U_1 having all source buses, U_2 having all other buses
+[I_1] = [ K  L ] [U_1]
+[I_2]   [ L' M ] [U_2]
+
+I_2 = MU_2 + L'U_1
+MU_2 = I_2 - L'U_1 = RHS
+
+
+Steps:
+Initialize U_2 with flat start and phase shifts accounted
+Add source admittance to Y bus matrix
+while maximum deviation > error tolerance
+    Calculate I_2 with U_2 of previous iteration as per load/gen types.
+    Solve MU_2 = RHS using prefactorization.
+    Find maximum deviation in voltage
+    Update U_2
+(Invalidate prefactorization if parameters change, ie y bus values changes)
+
+*/
+
+#include "../calculation_parameters.hpp"
+#include "../exception.hpp"
+#include "../power_grid_model.hpp"
+#include "../three_phase_tensor.hpp"
+#include "../timer.hpp"
+#include "block_matrix.hpp"
+#include "bsr_solver.hpp"
+#include "iterative_pf_solver.hpp"
+#include "y_bus.hpp"
+
+namespace power_grid_model {
+
+// hide implementation in inside namespace
+namespace math_model_impl {
+
+// solver
+template <bool sym>
+class IterativecurrentPFSolver : public IterativePFSolver<sym, IterativecurrentPFSolver<sym>> {
```

Comment:
Please use CamelCase

So class name should be `IterativeCurrentPFSolver`, with capital `C` for `Current`.